### ISSUE TITLE
HNC: cannot delete namespaces in 1.14

### ISSUE THREAD

**Issue by adrianludwin** [kubernetes-sigs/multi-tenancy#889]
Repro at head:

```
$ k hns create sub -n default
$ k delete subns sub -n default
```
Log messages show that the webhook is failing because we can't decode the old object. We should be skipping this.

If it turns out that this is actually working in 0.5, then we can change the milestone back to 0.6.

**Comment by adrianludwin** [comment]
This is due to the new object validator; the 0.5 validator didn't check deletions at all.

_Why_ the object validator is being invoked on a namespace with no objects in it is somewhat of a mystery to me, but this definitely works on 1.16 and definitely does _not_ work on 1.14. Since the code is obviously wrong, I'm going to go ahead and fix it.

**Comment by yiqigao217** [comment]
Do you mean that the `cannot decode` is in the object validator but not namespace validator? We added 1.14 backward compatibility (allow deleting in 1.14) in namespace validator. To fix this issue, is it that we need to also add it to object validator?

**Comment by adrianludwin** [comment]
Yeah. I don't understand it. Our namespace and anchor validators already handle a nil OldObject but the object validators didn't. This is a bug that needs to be fixed regardless, but I don't understand why the code was invoked when there were _no_ objects at all.

**Comment by yiqigao217** [comment]
Not sure if you've already noticed. Even deleting a full namespace (not subnamespace) doesn't work in 1.14 using the head before your fix #890 .
